test/functional/apps/dashboard/_view_edit.js

```diff
@@ -173,9 +174,12 @@ export default function ({ getService, getPageObjects }) {
           // was loaded initially
           await PageObjects.dashboard.loadSavedDashboard(dashboardName);
           await PageObjects.dashboard.clickEdit();
-
-          const originalFilters = await PageObjects.dashboard.getFilters();
-
+          const originalFilters = await retry.try(async () => {
+            return PageObjects.dashboard.getFilters().then(resp => {
+              if (!resp.length) throw new Error('expected filters');
+              return resp;
+            });
+          });
```

Comment:
Since everything else is using await, could maybe follow suit and do:
```
// Try until we get filters back to avoid a potential timing issue.
// See https://github.com/elastic/kibana/issues/11339 for more info.
const originalFilters = await retry.try(async () => {
  const filters = await PageObjects.dashboard.getFilters();
  if (!filters.length) throw new Error('expected filters');
  return filters;
});
```

Take it or leave it, just personal preference that it looks better to my eyes. Thanks for (hopefully) fixing this!